### PR TITLE
[json-rpc][pvr] Recordings: add fields season, episode and showtitle

### PR DESCRIPTION
**Pull request by flubshi**
## Description
This adds fields _season_, _episode_ and _showtitle_ to the JSON RPC response for recordings. If a recording is not a tv show, an empty string for showtitle and 0 for episode/season are returned.

## Motivation and Context
Using the web interface or an android remote applications, I noticed that for recordings that are tv shows, only the show name is returned by the API but not the tvshow title. This makes it impossible to distinguish between the episodes (without reading the plot) and reduces the user experience.
A next step would be to extend the recordings view of the web interface skins to display season/episode and showtitle.

## How Has This Been Tested?
I compiled kodi locally and used curl to retrieve the recordings:
`curl 'http://127.0.0.1:8080/jsonrpc?RecordingCollection' -H 'Content-Type: application/json; charset=UTF-8' -H 'Accept: application/json, text/javascript, */*; q=0.01' -H 'X-Requested-With: XMLHttpRequest'  --data-binary '{"jsonrpc":"2.0","method":"PVR.GetRecordings","id":"1554552409607","params":{"properties":["file","title","season","episode","showtitle"],"limits":{"start":0}}}'`

Which returns the fields as expected (extract shows a movie and a tv show):
`{"id":"1554552409607","jsonrpc":"2.0","result":{"limits":{"end":9,"start":0,"total":9},"recordings":[{"episode":0,"file":"pvr://recordings/tv/active/Detektiv Conan: Der dunkelste Albtraum/Detektiv%20Conan%3a%20Der%20dunkelste%20Albtraum (2016), TV, 20190329_181500, 40773759.pvr","label":"Detektiv Conan: Der dunkelste Albtraum","recordingid":11,"season":0,"showtitle":"","title":"Detektiv Conan: Der dunkelste Albtraum"},[...],{"episode":20,"file":"pvr://recordings/tv/active/Elementary/Elementary s06e20 (2018)%20Blinde%20Wut, TV, 20190405_181500, 41017361.pvr","label":"Elementary","recordingid":18,"season":6,"showtitle":"Blinde Wut","title":"Elementary"}]}}`

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/interfaces/json-rpc/schema/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-JSONRPC_VERSION 10.3.0
+JSONRPC_VERSION 10.3.1
```
Comment (DaveTBlake) [inline xbmc/interfaces/json-rpc/schema/version.txt]: This makes a non-breaking change - adding fields at end - hence needs to be a bump to 10.4.0

**Review by DaveTBlake — CHANGES_REQUESTED** [review]

File: xbmc/interfaces/json-rpc/schema/version.txt (hunk 1 above)
Comment (flubshi) [reply xbmc/interfaces/json-rpc/schema/version.txt]: Changed. Thanks!

**Review by DaveTBlake — APPROVED** [review]

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by ksooo** [comment]
Do not merge, don't want to have it like this, but  no time to elaborate atm.

**Review by ksooo — CHANGES_REQUESTED** [review]
Rename "showtitle" to "episodename", please.

**Comment by flubshi** [comment]
> Rename "showtitle" to "episodename", please.

Done. I agree that _episodename_ ismore intuitive. Thanks!

**Comment by ksooo** [comment]
Can you do me a favor and rename "m_strShowTitle" to "m_strEpisodeName"?

**Comment by flubshi** [comment]
> Can you do me a favor and rename "m_strShowTitle" to "m_strEpisodeName"?

Sure. But please re-check if I chose the right place for variable declaration.

File: xbmc/pvr/recordings/PVRRecording.cpp [hunk 2]
```diff
@@ -142,7 +142,7 @@ bool CPVRRecording::operator ==(const CPVRRecording& right) const
        m_strDirectory       == right.m_strDirectory &&
        m_strFileNameAndPath == right.m_strFileNameAndPath &&
        m_strTitle           == right.m_strTitle &&
-       m_strShowTitle       == right.m_strShowTitle &&
+	   m_strEpisodeName     == right.m_strEpisodeName &&
```
Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecording.cpp]: Please correct the indentation.

File: xbmc/pvr/recordings/PVRRecording.h [hunk 3]
```diff
@@ -69,6 +69,7 @@ namespace PVR
     std::string   m_strThumbnailPath; /*!< thumbnail path */
     std::string   m_strFanartPath;    /*!< fanart path */
     unsigned      m_iRecordingId;     /*!< id that won't change while xbmc is running */
+    std::string   m_strEpisodeName;   /*!< name of the episode */
```
Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecording.h]: Please remove the showtitle variable in favor to the episodename variable. The former is not needed anymore.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/pvr/recordings/PVRRecording.h (hunk 3 above)
Comment (flubshi) [reply xbmc/pvr/recordings/PVRRecording.h]: I think m_strShowTitle is defined in another file, in VideoInfoTag.h. But I am not sure if it makes sense to replace it everwhere.
In RecentlyAddedJob.cpp line 103/4 currently m_strShowTitle is assigned to ShowTitle and m_strTitle is assigned to EpisodeTitle. Replacing m_strShowTitle here seems to be wrong.

Or is m_strShowTitle independent from m_strShowTitle in recordings?

**Comment by ksooo** [comment]
No new features for v18.

File: xbmc/pvr/recordings/PVRRecording.h (hunk 3 above)
Comment (ksooo) [reply xbmc/pvr/recordings/PVRRecording.h]: Oh well, my fault. `m_strShowTitle` is defined in `CVideoInfoTag`from which `CPVRRecording`is derived. I overlooked this.

More interesting: CVideoInfoTag already exposes: 
* `m_strShowTitle` as property `showtitle`. 
* `m_iSeason`as property `season`
* `m_iEpisode` as property `episode`

As above `CVideoInfoTag`members get filled by `CPVRRecording`, every pvr recording already should expose the episode name, episode number and season number via its base class. If this is working as designed, there is no need to extend `CPVRRecording` which would render this PR "useless" as it adds no new functionality.

**Comment by ksooo** [comment]
Not sure this PR is actually adding anything new. See https://github.com/xbmc/xbmc/pull/15869#discussion_r274286491

**Comment by ksooo** [comment]
Maybe the actual problem to solve is that the fact that `CPVRRecording` is derived from `CPVRRecording` and thus json `PVR.Details.Recording` and json `PVR.Fields.Recording` inherit all video info tag properties is not reflected properly in our json schema?

I'm not an expert, but maybe this https://github.com/xbmc/xbmc/pull/15869/files#diff-1bb266d43e6d1a19799e53fcf6371260R1115 and this https://github.com/xbmc/xbmc/pull/15869/files#diff-1bb266d43e6d1a19799e53fcf6371260R1125 is wrong? @DaveTBlake

**Comment by ksooo** [comment]
Seems, that we are manually adding selected video info tag properties to json PVR.*.Recording types instead of getting everything via inheritance. No idea whether this is a good approach, but if it is, @flubshi could remove everything from the PR except the changes to `types.json` (and the API version bump ofc) and the three props in question should be available for PVR.*.Recording. No additional code changes needed.

**Comment by rmrector** [comment]
The video library JSON-RPC interface defines movies and episodes and other media types separately, but they are all `CVideoInfoTag` internally. I like this behavior, many of the fields in `CVideoInfoTag` aren't useful in each of the media types; only exposing the useful ones simplifies the API. I'm not a PVR user or developer but it makes sense to me to do the same for recordings.

**Comment by flubshi** [comment]
> [..] No additional code changes needed.

You are right! I runtime tested and it seems to work. Thanks ksooo!

**Review by ksooo — APPROVED** [review]

**Comment by DaveTBlake** [comment]
@ksooo sorry for late reply, but yes as @rmrector explained I think the schema is correct.

As a non-breaking change to the API and minimum risk I am also happy to see this relased in v18, but as it relates to PVR it is entirely your choice.

**Comment by ksooo** [comment]
You know my POW on this : fixes are fine for  v18, everything else is not.

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**